extensions/sovity-messenger/src/main/java/de/sovity/edc/extension/messenger/api/MessageHandlerRegistry.java

```diff
@@ -0,0 +1,78 @@
+/*
+ *  Copyright (c) 2024 sovity GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       sovity GmbH - initial API and implementation
+ */
+
+package de.sovity.edc.extension.messenger.api;
+
+import lombok.val;
+import org.eclipse.edc.spi.EdcException;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.function.Function;
+
+/**
+ * The component where to register message handlers when using the {@link SovityMessenger}.
+ */
+public interface MessageHandlerRegistry {
```

Comment:
inline / have one unit, you could  move it to the root level, if you don't have "api/impl" duality anymore